metaflow/includefile.py

```diff
@@ -33,15 +66,25 @@ def __call__(self):
         self._logger(
             'Including file %s of size %d%s %s' % (self._path, sz, unit[pos], extra))
         if self._is_text:
-            return io.open(self._path, mode='rt', encoding=self._encoding).read()
+            return io.open(path, mode='rt', encoding=self._encoding).read()
         try:
-            return io.open(self._path, mode='rb').read()
+            return io.open(path, mode='rb').read()
         except IOError:
             # If we get an error here, since we know that the file exists already,
             # it means that read failed which happens with Python 2.7 for large files
             raise MetaflowException('Cannot read file at %s -- this is likely because it is too '
                                     'large to be properly handled by Python 2.7' % self._path)
 
+    def __call__(self):
+        local_file = None
+        if self._path.startswith("s3://"):
+            local_file = self.handle_s3_file()
+        try:
+            return self.handle_local_file(local_file)
```

Comment:
To improve readability slightly I think you could use kwarg here and say `override_path=local_file`; so its clearer that we want to override only for S3 case.
Otherwise the function uses `self._path` for knowing the local file path; which was non-obvious to me until I read L:54-55.